torchvision/transforms/v2/functional/_utils.py

```diff
@@ -104,78 +114,32 @@ def _get_kernel(dispatcher, input_type):
             elif cls in registry:
                 return registry[cls]
 
-        # Note that in the future we are not going to return a noop here, but rather raise the error below
-        return _noop
+    if allow_passthrough:
+        return _passthrough
 
     raise TypeError(
-        f"Dispatcher {dispatcher} supports inputs of type torch.Tensor, PIL.Image.Image, "
-        f"and subclasses of torchvision.datapoints.Datapoint, "
+        f"Dispatcher F.{dispatcher.__name__} supports inputs of type {registry.keys()}, "
         f"but got {input_type} instead."
     )
 
 
-# Everything below this block is stuff that we need right now, since it looks like we need to release in an intermediate
-# stage. See https://github.com/pytorch/vision/pull/7747#issuecomment-1661698450 for details.
-
-
-# In the future, the default behavior will be to error on unsupported types in dispatchers. The noop behavior that we
-# need for transforms will be handled by _get_kernel rather than actually registering no-ops on the dispatcher.
-# Finally, the use case of preventing users from registering kernels for our builtin types will be handled inside
-# register_kernel.
-def _register_explicit_noop(*datapoints_classes, warn_passthrough=False):
-    """
-    Although this looks redundant with the no-op behavior of _get_kernel, this explicit registration prevents users
-    from registering kernels for builtin datapoints on builtin dispatchers that rely on the no-op behavior.
-
-    For example, without explicit no-op registration the following would be valid user code:
-
-    .. code::
-        from torchvision.transforms.v2 import functional as F
-
-        @F.register_kernel(F.adjust_brightness, datapoints.BoundingBox)
-        def lol(...):
-            ...
-    """
-
+def _register_temporary_passthrough_kernels_internal(*datapoints_classes):
```

Comment:
wait, do we need this, now that we let the transforms pass-through? I thought we were doing this originally in case we didn't have time to properly raise before the release. Now that all the transforms properly pass-through, shouldn't we just raise in those kernels?

Reply:
I was under the impression that we want to pass-through on the dispatchers as well. However, your argument makes more sense. I'll fix.